### PR TITLE
Added new API calls and update Demo app

### PR DESCRIPTION
**Pull request by Vermillion828**
Comments, attachments methods, etc.

### CODE REVIEW

File: DribbbleSDK.podspec [hunk 1]
```diff
@@ -23,8 +23,8 @@ Uses latest Dribbble HTTP API !
     s.license          = { :type => 'MIT', :file => 'LICENSE' }
     s.authors           = { 'Agilie' => 'info@agilie.com' }
     s.source           = { :git => "https://github.com/agilie/dribbble-ios-sdk.git",
-                            :tag => s.version.to_s
-                            :branch => "public"
+                            :tag => s.version.to_s,
+                            :branch => "dev1"
```
Comment (dmitry-salnikov) [inline DribbbleSDK.podspec]: change to 'public'

File: DribbbleSDK/Client/DRApiClient.h [hunk 2]
```diff
@@ -37,45 +37,60 @@ extern void DRLog(NSString *format, ...);
...
+// comments
+
 - (void)uploadCommentWithShot:(NSNumber *)shotId withBody:(NSString *)body responseHandler:(DRResponseHandler)responseHandler;
 - (void)updateCommentWith:(NSNumber *)commentId forShot:(NSNumber *)shotId withBody:(NSString *)body responseHandler:(DRResponseHandler)responseHandler;
 - (void)deleteCommentWith:(NSNumber *)commentId forShot:(NSNumber *)shotId responseHandler:(DRResponseHandler)responseHandler;
-
 - (void)loadCommentsWithShot:(NSNumber *)shotId params:(NSDictionary *)params responseHandler:(DRResponseHandler)responseHandler;
 - (void)loadCommentWith:(NSNumber *)commentId forShot:(NSString *)shotId responseHandler:(DRResponseHandler)responseHandler;
-- (void)loadLikesWithComment:(NSNumber *)commentId forShot:(NSString *)shotId params:(NSDictionary *)params responseHandler:(DRResponseHandler)responseHandler;
+- (void)likeWithComment:(NSNumber *)commentId forShot:(NSString *)shotId responseHandler:(DRResponseHandler)responseHandler;
```
Comment (dmitry-salnikov) [inline DribbbleSDK/Client/DRApiClient.h]: make all the "id" properties type NSNumber

File: DribbbleSDK/Client/DRApiClient.m [hunk 3]
```diff
@@ -146,21 +145,30 @@ - (AFHTTPRequestOperation *)createRequestWithMethod:(NSString *)method requestTy
             responseHandler([weakSelf mappedDataFromResponseObject:responseObject modelClass:modelClass]);
         }
     } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
+        if ([operation.response statusCode] == 404) {
```
Comment (dmitry-salnikov) [inline DribbbleSDK/Client/DRApiClient.m]: extract to constant

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 4]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+            if (response.object) {
+                if ([response.object isKindOfClass:[NSArray class]]) {
+                    if ([response.object count]) {
+                        DRShot *shot = [response.object firstObject];
+                        if(shot) {
+                            self.delegate.shot = shot;
+                            [self.apiClient loadCommentsWithShot:shot.shotId params:@{} responseHandler:^(DRApiResponse *response) {
+                                if (response.object) {
+                                    for (DRComment *comment in response.object) {
+                                        if (([comment.body isEqualToString:@"<p>API test updated comment</p>"] || [comment.body isEqualToString:@"<p>API test comment</p>"]) &&
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: remove this

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 5]
```diff
@@ -37,7 +38,7 @@
 @interface MainViewController () <UITableViewDelegate, UITableViewDataSource>
 
 @property (strong, nonatomic) DRApiClient *apiClient;
-
+@property (strong, nonatomic) AppDelegate *delegate;
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m:41]: remove data model properties from delegate and store them in this view controller

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 6]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+        [self loadMockData];
+    }
+    
+    self.apiCallWrappers = [ApiCallFactory demoApiCallWrappers];
+    [self.tableView reloadData];
+}
+
+- (void)loadMockData {
+    [self.apiClient loadUserInfoWithResponseHandler:^(DRApiResponse *response) {
+        if (response.object) {
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: change to [response.object isKindOfClass..

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 7]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+            self.delegate.user = response.object;
+        }
+        
+        [self.apiClient loadShotsWithUser:self.delegate.user.userId params:@{} responseHandler:^(DRApiResponse *response) {
+            if (response.object) {
+                if ([response.object isKindOfClass:[NSArray class]]) {
+                    if ([response.object count]) {
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: merge conditions from lines 121-123

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 8]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+                        DRShot *shot = [response.object firstObject];
+                        if(shot) {
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: change to isKindOfClass condition

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 9]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+                                    for (DRComment *comment in response.object) {
+                                        if (([comment.body isEqualToString:@"<p>API test updated comment</p>"] || [comment.body isEqualToString:@"<p>API test comment</p>"]) &&
+                                            comment.user.userId == self.delegate.user.userId) {
+                                            self.delegate.comment = comment;
+                                            self.apiCallWrappers = [ApiCallFactory demoApiCallWrappers];
+                                            [self.tableView reloadData];
+                                        }
+                                    }
+                                    [self.apiClient loadAttachmentsWithShot:shot.shotId params:@{} responseHandler:^(DRApiResponse *response) {
+                                        if (response.object && [response.object isKindOfClass:[NSArray class]]) {
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: remove first part from condition (second part includes it)

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 10]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+                                            DRShotAttachment *attachment = [response.object lastObject];
+                                            self.delegate.attachment = attachment;
+                                            self.apiCallWrappers = [ApiCallFactory demoApiCallWrappers];
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: remove copypaste (about apicallwrappers and reloaddata)

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 11]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+                                        if (([comment.body isEqualToString:@"<p>API test updated comment</p>"] || [comment.body isEqualToString:@"<p>API test comment</p>"]) &&
+                                            comment.user.userId == self.delegate.user.userId) {
+                                            self.delegate.comment = comment;
+                                            self.apiCallWrappers = [ApiCallFactory demoApiCallWrappers];
+                                            [self.tableView reloadData];
+                                        }
+                                    }
+                                    [self.apiClient loadAttachmentsWithShot:shot.shotId params:@{} responseHandler:^(DRApiResponse *response) {
+                                        if (response.object && [response.object isKindOfClass:[NSArray class]]) {
+                                            DRShotAttachment *attachment = [response.object lastObject];
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: please use the same getter for arrays with 1 object. preferred it the "firstObject"

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m [hunk 12]
```diff
@@ -94,15 +96,73 @@ - (void)setupApiClient {
...
+                                }
+                            }];
+                        }
+                    }
+                }
+            }
+        }];
+        
+        self.apiCallWrappers = [ApiCallFactory demoApiCallWrappers];
+        [self.tableView reloadData];
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: remove copypaste (see above)

File: DribbbleSDK.podspec (hunk 1 above)
Comment (dmitry-salnikov) [inline DribbbleSDK.podspec]: please update version to 0.0.3

File: DribbbleSDK/Client/DRApiClient.m [hunk 13]
```diff
@@ -146,21 +145,30 @@ - (AFHTTPRequestOperation *)createRequestWithMethod:(NSString *)method requestTy
...
     return operation;
 }
 
-- (void)runMultiPartRequestWithMethod:(NSString *)method params:(NSDictionary *)params data:(NSData *)data mimeType:(NSString *)mimeType responseHandler:(DRResponseHandler)responseHandler {
-    UIImage *image = [[UIImage alloc] initWithData:data];
-    CGSize imageSize = image.size;
-    NSAssert((imageSize.width == 400.f && imageSize.height == 300.f) || (imageSize.width == 800.f && imageSize.height == 600.f), kUploadImageSizeAssertionString);
-    NSAssert((data.length/1024.f/1024.f) <= kUploadFileBytesLimitSize, kUploadFileSizeAssertionString);
-    
+- (void)runMultiPartRequestWithMethod:(NSString *)method params:(NSDictionary *)params data:(NSData *)data mimeType:(NSString *)mimeType isAttachment:(BOOL)isAttachment responseHandler:(DRResponseHandler)responseHandler {
```
Comment (dmitry-salnikov) [inline DribbbleSDK/Client/DRApiClient.m]: refactor. add parameter "parameterName" in this method, remove isAttachment condition (the main point is: keep generic method generic). add two methods like: uploadAttachment and uploadFile, and these methods should call generic method with differrent value of parameterName

Comment (dmitry-salnikov) [inline DribbbleSDK/Client/DRApiClient.m]: one more thing: 
if (responseHandler) responseHandler([DRApiResponse responseWithError:userError]);
        } else {
            if (responseHandler) responseHandler([DRApiResponse responseWithError:error]);
- refactor; remove "else" condition and just replace error value with userError value

File: DribbbleSDK/Client/DRApiClient.m [hunk 14]
```diff
@@ -329,14 +341,30 @@ - (void)checkLikeWithComment:(NSNumber *)commentId forShot:(NSString *)shotId re
...
+}
+
 - (void)loadAttachmentsWithShot:(NSNumber *)shotId params:(NSDictionary *)params responseHandler:(DRResponseHandler)responseHandler {
-    [self runRequestWithMethod:[NSString stringWithFormat:kDRApiMethodShotAttachments, shotId] requestType:kHttpMethodGet modelClass:[DRShotAttachment class] params:params responseHandler:responseHandler];
+    [self runRequestWithMethod:[NSString stringWithFormat:kDRApiMethodShotAttachments, shotId] requestType:kHttpMethodGet modelClass:[DRShotAttachment class] params:nil responseHandler:responseHandler];
 }
 
 - (void)loadAttachmentWith:(NSNumber *)attachmentId forShot:(NSString *)shotId params:(NSDictionary *)params responseHandler:(DRResponseHandler)responseHandler {
-    [self runRequestWithMethod:[NSString stringWithFormat:kDRApiMethodAttachment, attachmentId, shotId] requestType:kHttpMethodGet modelClass:[DRShotAttachment class] params:params responseHandler:responseHandler];
+    [self runRequestWithMethod:[NSString stringWithFormat:kDRApiMethodAttachment, shotId, attachmentId] requestType:kHttpMethodGet modelClass:[DRShotAttachment class] params:nil responseHandler:responseHandler];
```
Comment (dmitry-salnikov) [inline DribbbleSDK/Client/DRApiClient.m:381]: remove params from method interface, and for all similar interfaces

File: DribbbleSDKExample/DribbbleSDKExample/ApiCallFactory.m [hunk 15]
```diff
@@ -26,7 +27,7 @@ + (ApiCallWrapper *)apiCallWrapperWithTitle:(NSString *)title selector:(SEL)sele
 }
 
 + (NSArray *)demoApiCallWrappers {
-    
+    AppDelegate *delegate = [AppDelegate delegate];
```
Comment (dmitry-salnikov) [inline DribbbleSDKExample/DribbbleSDKExample/ApiCallFactory.m]: remove data properties from delegate at all (and remove them from VC too). just make parametrized contructor for demoApiCallsWrappers

File: DribbbleSDK/Client/DRApiClient.h (hunk 2 above)
Comment (kalamaznik) [inline DribbbleSDK/Client/DRApiClient.h]: done

File: DribbbleSDK/Client/DRApiClient.m (hunk 3 above)
Comment (kalamaznik) [inline DribbbleSDK/Client/DRApiClient.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 6 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 7 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 8 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 9 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 10 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 11 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 12 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m]: done

File: DribbbleSDK/Client/DRApiClient.m (hunk 13 above)
Comment (kalamaznik) [inline DribbbleSDK/Client/DRApiClient.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/ApiCallFactory.m (hunk 15 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/ApiCallFactory.m]: done

File: DribbbleSDKExample/DribbbleSDKExample/MainViewController.m (hunk 5 above)
Comment (kalamaznik) [inline DribbbleSDKExample/DribbbleSDKExample/MainViewController.m:41]: done

**Comment by kalamaznik** [comment]
Added api methods (uploading attachments, comments, editing, work with bucket,) and update demo calls

**Pull request merged**